lib/bundle/brew_installer.rb

```diff
@@ -9,11 +9,18 @@ def initialize(name, options = {})
         @full_name = name
         @name = name.split("/").last
         @args = options.fetch(:args, []).map { |arg| "--#{arg}" }
+        @restart = options.fetch(:restart_service, false)
       else
         raise "Unable to install #{name}. Homebrew is not currently installed on your system"
       end
     end
 
+    def run
+      install_or_upgrade && restart
```

Comment:
Sorry i still don't get it. Can you make a quick pseudo what `self.install` and `install_or_upgrade` will be what you add a parameter like that. 
I think it will be like this https://gist.github.com/voanhduy1512/e6993637ca204f630539 but i feel it isn't as clean as current implement